### PR TITLE
Optional ExtHosts - improve codesplitting/treeshaking

### PR DESCRIPTION
**Pull request by CGNonofr**
based on https://github.com/CodinGame/monaco-vscode-api/pull/134

related to #136

It allows to only import some parts of a api.

It would allow for monaco-languageclient to only import what is required from the api by importing `vscode/ext-hosts/languageClient` only

(that's only 14 parts out of 26)

### CODE REVIEW

File: src/override/vs/workbench/browser/parts/editor/textEditor.weak.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  static [Symbol.hasInstance](instance) {
+      return AbstractTextEditor != null && instance instanceof AbstractTextEditor
+  }
+}
+
+export function setAbstractTextEditor(_AbstractTextEditor) {
+  AbstractTextEditor = _AbstractTextEditor
+}
+
+export { FakeAbstractTextEditor as AbstractTextEditor }
```
Comment (CGNonofr) [inline src/override/vs/workbench/browser/parts/editor/textEditor.weak.js:17]: @kaisalmen I didn't find any better way to do it, any other/more automatic idea?

Comment (kaisalmen) [reply src/override/vs/workbench/browser/parts/editor/textEditor.weak.js:17]: @CGNonofr I Don't have a better idea

**Review by kaisalmen — DISMISSED** [review]
LGTM. I need to test the full stack with `monaco-languageclient`, so the approval of #146 is the one that counts.

**Comment by CGNonofr** [comment]
> LGTM. I need to test the full stack with `monaco-languageclient`, so the approval of #146 is the one that counts.

Yeah sure, we'll only merge the last one when everything will be reviewed

**Pull request merged**